### ISSUE TITLE
Static file version information is missing in the real-time data feed

### ISSUE THREAD

**Issue by Gerriko** [google/transit#362]
I was developing an application that analyses GTFS-R data feed and last week I encountered an error where my results differed from the operator's real-time app. At first I thought it was my code, but after many hours of testing I discovered that the public transit operator had just updated the GTFS static files (e.g. trips.txt, stop_times.txt, routes.txt etc.) for the winter schedule.

I went back to the GTFS-R specification and I could not find any information about how static file versions are provided within the real-time data feed. I would have assumed that this vital piece of information would be recorded in the header section at least to allow apps an automated means to check that it is using the correct file versions.

Is this correct?

If so, maybe this needs changing, or have others implemented other automated steps to let your app know that the static files it is using need updating. How does one know how to handle the transition or swap over day. Does one just have to assume that the version you've just downloaded is the same one the GTFS-R is using etc.

**Comment by doconnoronca** [comment]
I have also had problems with some transit agencies where their published GTFS data does not match what is being used by their GTFS real time system. The LA Metro rail is in that state right now. Maybe adding a url and last update time to the GTFS real-time header would be useful.

**Comment by evantrillium** [comment]
might be worth revisiting #255 
Additionally, Trillium produces linked_datasets.txt as a means of translating GTFS-RT feed info from the static side as well. Having a codified, complementary language between the datasets/feeds would be ideal.

**Comment by skinkie** [comment]
This should be prioritized.

**Comment by gcamp** [comment]
Simplest way I see would be to add `feed_version` from `feed_info.txt` in the GTFS-rt header

**Comment by doconnoronca** [comment]
Adding this would help inform consumers when a new GTFS file is available and help coordinate between changes included in GTFS real-time and changes included in GTFS.

A GTFS update time (in POSIX time format) and a url to the feed being used would also be useful. Sometimes agencies have multiple GTFSs and it's not clear which one is to be used. San Francisco MUNI has this problem.

**Comment by stevenmwhite** [comment]
I agree that having a reference (link) to the schedule feed in the RT feed would be incredible helpful.

We come across this issue all the time, when our customer agencies have a RT and Schedule feed from us, but also have other schedule feeds floating around out there (from their planning software, most commonly, but sometimes from other sources or old sources as well).

One of the biggest problems we see is that a consumer will look at one of those other static feeds, our RT feed, and then things don't match,

**Comment by doconnoronca** [comment]
Here is a proposed addition to the FeedHeader message

```
// String that matches the feed_info.feed_version from the GTFS feed that the real
// time data is based on. Consumers can use this to identify which GTFS feed is
// currently active and when a new one is available to download.
optional string feed_version = 4;

// String containing an URL linking to the GTFS file. It should be the feed that is
// currently active.
// The URL should be a fully qualified URL that includes http:// or https://, and any
// special characters in the URL must be correctly escaped.
optional string GTFS_url = 5;
```

**Comment by doconnoronca** [comment]
So the next step would be for a producer to add this to their feed then I, as a consumer, can add support for it. Finally a pull request can then be created.
